sale_operating_unit/views/sale_view.xml

```diff
@@ -39,7 +39,23 @@
                     name="operating_unit_id"
                     widget="selection"
                     groups="operating_unit.group_multi_operating_unit"
+                    readonly="state not in ['draft', 'sent']"
                 />
+                <field
+                    name="operating_unit_id"
+                    invisible="1"
+                    groups="!operating_unit.group_multi_operating_unit"
+                />
+            </field>
+            <field name="analytic_account_id" position="attributes">
+                <attribute
+                    name="domain"
+                >["|",("operating_unit_ids","in", operating_unit_id),("operating_unit_ids","in", False)]</attribute>
```

Comment:
I face this warning:
`2024-06-18 09:18:06,328 141803 WARNING v17c_sale_operating_unit odoo.osv.expression: The domain term '('operating_unit_ids', 'in', False)' should use the '=' or '!=' operator. `

Reply:
on the other hand, you should keep the domain defined in python file
`['|', ('company_id', '=', False), ('company_id', '=', company_id)]`

Reply:
I fixed the domain itself. 
I also added the domain defined in the python file but I'm not sure it's useful. I mean the domain on the company_id is automatically added by the check_company

Reply:
yes, it's useful. you can define the domain in python file or in the view. in my understanding, the one from the view is higher priority than the one from python file. `check_company` attribute will be handled as a domain from python file, thus, will be ignored in this case.